### PR TITLE
Add unit tests for RenderStatePackager (close #62)

### LINKED ISSUE #62: Add unit tests for RenderStatePackager
Verify different edge cases: missing files, incorrect paths, wrong json files, missing objects, etc.

### PR DESCRIPTION
**Pull request by MikhailGorobets**

### CODE REVIEW

**Comment by TheMostDiligent** [comment]
Please rename .drsn to .json as .json is a universally recognized extensions and our files are just plain JSON files.

File: Tests/DiligentToolsTest/src/RenderStatePackager/RenderStatePackagerTest.cpp [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+    {
+        std::vector<std::string> InputFilePaths{"MissingShader.drsn"};
+        ASSERT_TRUE(pConverter->ParseFiles(InputFilePaths));
+
+        RefCntAutoPtr<IArchiver> pArchive;
+        pArchiveFactory->CreateArchiver(pEnvironment->GetSerializationDevice(), &pArchive);
+
+        ASSERT_FALSE(pConverter->Execute(pArchive));
+        pConverter->Flush();
```
Comment (TheMostDiligent) [inline Tests/DiligentToolsTest/src/RenderStatePackager/RenderStatePackagerTest.cpp:145]: I suggest renaming Flush to Clear or Reset.
Flush typically means writing out some buffer that was written, but this method simply resets the converter state.

**Review by TheMostDiligent — CHANGES_REQUESTED** [review]
